File: javascript/easy/base7.js
/**
 * @param {number} num
 * @return {string}
 */
var convertToBase7 = function(num) {
    if(num === 0) return "0";
    let res = "", abs = Math.abs(num);
    while(abs > 0) {
        res = (abs%7) + res;
        abs = parseInt(abs/7);
    }
    return (num < 0) ? ("-" + res) : res;
};

File: ruby/easy/base_7.rb
# @param {Integer} num
# @return {String}
def convert_to_base7(num)
    return "0" if num == 0
    abs = num.abs
    res = ""
    while abs > 0
        res = (abs % 7).to_s + res
        abs /= 7
    end
    return (num < 0) ? ("-" + res) : res
end
